### PR TITLE
WD-8399 - Create add secrets panel

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Create components for the add secrets form and panel (submitting the form will be handled in [WD-8546](https://warthogs.atlassian.net/browse/WD-8546)).

## QA

- Go to the secrets tab.
- Click "Add secret"
- The panel should open and display a form.

## Details

https://warthogs.atlassian.net/browse/WD-8399

## Screenshots

Screenshot 2024-01-30 at 4 05 36 pm

[WD-8546]: https://warthogs.atlassian.net/browse/WD-8546?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/panels/AddSecretPanel/AddSecretPanel.test.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { screen } from "@testing-library/react";
+
+import { renderComponent } from "testing/utils";
+
+import AddSecretPanel, { TestId } from "./AddSecretPanel";
+
+describe("AddSecretPanel", () => {
+  it("renders", async () => {
```
Comment (huwshimi) [inline src/panels/AddSecretPanel/AddSecretPanel.test.tsx:8]: This is just a placeholder test for when the functionality is added in https://warthogs.atlassian.net/browse/WD-8546.

File: src/panels/AddSecretPanel/AddSecretPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        <Formik<FormFields>
+          initialValues={{
+            content: "",
+            description: "",
+            expiryTime: "",
+            isBase64: false,
+            label: "",
+            rotatePolicy: RotatePolicy.NEVER,
+          }}
+          onSubmit={(values) => {
```
Comment (huwshimi) [inline src/panels/AddSecretPanel/AddSecretPanel.tsx:74]: This is where the secret will be created in https://warthogs.atlassian.net/browse/WD-8546.

File: src/panels/AddSecretPanel/Fields/Fields.test.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { screen } from "@testing-library/react";
+import { Formik } from "formik";
+
+import { renderComponent } from "testing/utils";
+
+import Fields from "./Fields";
+
+describe("Fields", () => {
+  it("renders", async () => {
```
Comment (huwshimi) [inline src/panels/AddSecretPanel/Fields/Fields.test.tsx:9]: This is just a placeholder test for when the functionality is added in https://warthogs.atlassian.net/browse/WD-8546.

File: src/panels/Panels.tsx [hunk 4]
```diff
@@ -25,6 +26,8 @@ export default function Panels() {
         return <ConfigPanel />;
       case "audit-log-filters":
         return <AuditLogsFilterPanel />;
+      case "add-secret":
+        return <AddSecret />;
```
Comment (vladimir-cucu) [inline src/panels/Panels.tsx:30]: I see that we use the "Panel" suffix for the names of all the previous panels. For consistency reasons, maybe it makes sense to name this something like "AddSecretPanel"?

**Review by vladimir-cucu — APPROVED** [review]
Looks good to me and works as expected locally! :rocket:

File: src/panels/Panels.tsx (hunk 4 above)
Comment (huwshimi) [reply src/panels/Panels.tsx]: Done.

**Pull request merged**